### ISSUE TITLE
Creation of dynamic property Cylancer\Loginviaemail\Services\EMailFrontendUserAuthenticationService::$info is deprecated

### ISSUE THREAD

**Issue by ps-devstudio** [docmcfly/loginViaEmail#22]
TYPO3 V12.4.15 - loginviaemail 4.0.0

![Bildschirmfoto 2024-05-29 um 08 26 59](https://github.com/docmcfly/loginViaEmail/assets/83782511/ce414f49-4e0b-4068-902a-7febcb3a8d6e)

Possible solution:
add 
`   /** @var array */
    protected $info = [];` 
to line 27

https://github.com/docmcfly/loginViaEmail/blob/922b870c813e84b50b40667ed379f9b998b3540b/Classes/Services/EMailFrontendUserAuthenticationService.php#L27

**Comment by docmcfly** [comment]
Which PHP are you using? 

I cannot reproduce the error. Also I can't find access to $this->info. So why should I create the member variable separately. This is not clear to me. 

I use PHP-Version: 8.1.18 on my development and test  system. ( and Typo3 12.4.15 too) 

best regards

**Comment by ps-devstudio** [comment]
Yes. PHP-version-problem. I used 8.2, when I change to 8.1 problem is gone.

Thank you!

**Comment by docmcfly** [comment]
I have also tried PHP version 8.2.5. But I could not reproduce the error. 

As long as I cannot reproduce the error, I will not make any code changes. 

Sorry...  

PS: Thank you for your issue and using of my extension.

**Comment by slavlee** [comment]
I have the same problem with

- TYPO3 13.4.12
- loginviaemail 5.0.2
- PHP 8.2.28
- running on ddev

During the Authentification process the public property info is expected, because it is set by TYPO3. Solution is to create

public array $info = [] in EMailFrontendUserAuthenticationService

**Comment by docmcfly** [comment]
A northern German 'Moin' for you (or simply hello),

Thank you for your post. Unfortunately, I cannot
a) reproduce your error. The tests allow a login via email and username, and nothing happens in case of incorrect input.
b) After analyzing the code, I cannot find any $info or that it is being populated by TYPO3 anywhere.

For this reason, I see no justification for adding a public variable to the code without a real necessity.

The issue remains closed.

**Comment by slavlee** [comment]
Hi,

the GeneralUtility::makeInstanceService (line 3157) does assign on the parent service object a value to the public info property. And in my case it breaks there see error message:

Core: Exception handler (WEB): Uncaught TYPO3 Exception: Cannot access protected property Cylancer\Loginviaemail\Services\EMailFrontendUserAuthenticationService::$info | Error thrown in file /var/www/html/vendor/typo3/cms-core/Classes/Utility/GeneralUtility.php in line 3157. Requested URL: https://verlagderautoren-website.ddev.site/theater-verlag/interner-bereich/login.html?tx_felogin_login%5Baction%5D=login&tx_felogin_login%5Bcontroller%5D=Login&cHash=c057979448dc3adba781390d595c5194

**Comment by docmcfly** [comment]
I suspect there is a misunderstanding here. While the class name does contain 'Service,' it is not actually a service by TYPO3's definition. Therefore, there is no $info.

I would like to know in what context '#makeInstanceService()' is called. Is there a complete stack trace?

For clarification: The extension loginViaMail only needs to be installed, and then it works with the usual FrontendLogin. It is registered as another FrontendUserAuthentication and integrates into the series of other FrontendUserAuthentications. It is not a truly standalone service.

More details on the topic:
https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Authentication/AuthenticationService/Index.html

**Comment by docmcfly** [comment]
Addendum: Something is different on your end, as it works for many others.

**Comment by slavlee** [comment]
In the ext_localconf.php it is registered as  TYPO3 Service with \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addService(..) and therefore it is instantiated with makeInstanceService inside the AbstractUserAuthenticiation by getAuthServices.

If you give me your email address then I can send you a screenshot of the full error stack.

**Comment by docmcfly** [comment]
Thank you very much for the suggestion. I finally had some time to delve deeper into the topic.

It seems that my previous service is not a real service.

However, there is also an AbstractAuthenticationService. By inheriting from this, I fulfill all the properties of a service. Therefore, I have switched the logic to this AbstractAuthenticationService.

I have checked in the new version in the development branch. I would be very grateful if you could test this development version to see if there is still a problem with '#makeServiceInstance()'.

Thank you very much for your suggestion and your persistence.

I look forward to your feedback... Clemens

**Comment by docmcfly** [comment]
@slavlee 
Can I withdraw my correction and close the issue?
